src/modules/dashboard/gameDev/gameDevHandlers/achievementGameDevHandlers.ts

```diff
@@ -30,6 +32,22 @@ export const getAchievementsHandler: Lifecycle.Method = async (request, h) => {
   }
 };
 
+export const getAchievementProgressHandler: Lifecycle.Method = async (request, h) => {
+  if (!request.params.achievementId) {
+    throw Boom.notFound('Invalid achievementId');
+  }
+  const achievement = await findAchievementById(request.params.achievementId);
+
+  if (!achievement) {
+    throw Boom.notFound('achievement not found');
+  }
+  const achievementUnlockedList = await getAchievementUnlockedFromAchievement(achievement);
```

Comment:
getAchievementUnlockedFromAchievement has a default limit param set to 10. you should probably call this function using  queryString.limit params as well . otherwise you will always get a max of 10 results